examples/example.rs

```diff
@@ -0,0 +1,127 @@
+extern crate neat_gru;
+
+use itertools::Itertools;
+use neat_gru::game::Game;
+use neat_gru::neural_network::nn::NeuralNetwork;
+use neat_gru::topology::topology::Topology;
+use neat_gru::train::train::Train;
+struct Player {
+    pub net: NeuralNetwork<f64>,
+}
+
+impl Player {
+    pub fn new(net: NeuralNetwork<f64>) -> Player {
+        Player { net }
+    }
+    /// Runs all the inputs and calculates the outputs
+    fn run(&mut self) -> f64 {
+        // Get the inputs
+        let inputs = XOR::get_inputs();
+        // Calculate a score for every input
+        let outputs: Vec<f64> = inputs.iter().map(|i| self.net.compute(i)[0]).collect();
+        let mut scores: Vec<f64> = vec![];
+        for idx in 0..(inputs.len() - 1) {
+            scores.push(compute_score(&inputs[idx], outputs[idx]));
+        }
+        // And return the sum of the scores
+        scores.iter().sum()
+    }
+}
+
+struct Simulation {
+    players: Vec<Player>,
+}
+
+impl Simulation {
+    pub fn new() -> Simulation {
+        Simulation {
+            players: Vec::new(),
+        }
+    }
+}
+
+struct XOR {
+    inputs: Vec<Vec<f64>>,
+    answers: Vec<Vec<f64>>,
+}
+
+impl XOR {
+    pub fn new() -> Self {
+        XOR {
+            inputs: vec![
+                vec![0.0, 0.0],
+                vec![1.0, 1.0],
+                vec![1.0, 0.0],
+                vec![0.0, 1.0],
+            ],
+            answers: vec![vec![0.0], vec![0.0], vec![1.0], vec![1.0]],
+        }
+    }
+    fn get_inputs<'a>() -> &'a [[f64; 2]; 4] {
+        &[[0.0, 0.0], [1.0, 1.0], [1.0, 0.0], [0.0, 1.0]]
+    }
+}
+
+/// Computes the score with given inputs and one output
+fn compute_score(inputs: &[f64], output: f64) -> f64 {
+    // https://en.wikipedia.org/wiki/XOR_gate
+    // Returns 1.0 for a wrong output and 0.0 for a right output. Should be used as a score
+    let inputs: Vec<&f64> = inputs.into_iter().collect_vec();
+    if (*inputs[0] == 0.0 && *inputs[1] == 1.0) || (*inputs[0] == 1.0 && *inputs[1] == 0.0) {
```

Comment:
Can be reduced to *inputs[0] ^ *inputs[1] == output

Reply:
Yup. Though i should convert them to a boolean first as well as the output which would probably also take care of the Rounding issue